dask/bag/core.py

```diff
@@ -183,6 +183,42 @@ def finalize_item(results):
     return results[0]
 
 
+def merge_kwargs(dsk, kwargs):
+    """ Merges dask values from kwargs with dsk.
+
+    Returns the merged dsk and a graph expression useful for use with apply in
+    a dask graph to pass the final computed **kwargs to a function.
+
+    Currently only dask.bag.Item and python literal values are supported.
+    """
+    kw_pairs = []
+    for key, val in iteritems(kwargs):
+        if isinstance(val, Item):
+            dsk = merge(dsk, val.dask)
+            val = val.key
+        # TODO elif isinstance(val, Value):
```

Comment:
I've intentionally left this (and e.g. dask.dataframe.Scalar) out for now simply because I've had enough surprises of my own when crossing scheduler boundaries to be wary of just allowing this by default.  However there is quite a bit of missing sugar for users to tell dask "no really please do it" explicitly; things like:
- `bag.item.to_imperative`
- `imperative.value.to_item`
- `dataframe.scalar.to_` anything

Despite what regret there may be about having multiple types for "a scalar computation value", I think that we should at least add commuting methods between all three (that I've found so far at least) competing types for such a thing ;-)